Framework/jmaqs-appium/src/main/java/com/magenic/jmaqs/appium/AppiumUtilities.java

```diff
@@ -0,0 +1,236 @@
+   * @return Path to Screenshot.
+   */
+  public static String captureScreenshot(AppiumDriver<WebElement> appiumDriver,
+      AppiumTestObject testObject, String directory, String fileNameWithoutExtension) {
+    File tempFile = appiumDriver.getScreenshotAs(OutputType.FILE);
+    String path = Paths.get(directory, fileNameWithoutExtension + ".png").normalize().toString();
+
+    // Make sure the directory exists
+    try {
+      Path tempPath = new File(directory).toPath();
+      if (!tempPath.toFile().isDirectory()) {
+        Files.createDirectories(tempPath);
+      }
+    } catch (IOException exception) {
+      testObject.getLog()
+          .logMessage(MessageType.ERROR, "Failed to create directories: " + exception.getMessage());
+    }
+
+    // Try to copy the temporary file to desired file path
+    try {
+      Files.copy(tempFile.toPath(), new File(path).toPath(), StandardCopyOption.COPY_ATTRIBUTES);
+    } catch (IOException exception) {
+      testObject.getLog().logMessage(MessageType.ERROR,
+          String.format("Screenshot error: %s", exception.getMessage()));
+    }
+
+    testObject.addAssociatedFile(path);
+    return path;
+  }
+
+  /**
+   * To capture a page source during execution.
+   * Default parameter appendName of empty string.
+   *
+   * @param appiumDriver The Appium Driver
+   * @param testObject   The Appium Test Object
+   * @return True if saving page source is successful, otherwise false
+   */
+  public static boolean savePageSource(AppiumDriver<WebElement> appiumDriver,
+      AppiumTestObject testObject) {
+    return savePageSource(appiumDriver, testObject, "");
+  }
+
+  /**
+   * To capture a page source during execution.
+   *
+   * @param appiumDriver The Appium Driver
+   * @param testObject   The Appium Test Object
+   * @param appendName   Appends a name to the end of a filename
+   * @return True if saving page source is successful, otherwise false
+   */
+  public static boolean savePageSource(AppiumDriver<WebElement> appiumDriver,
+      AppiumTestObject testObject, String appendName) {
+    try {
+      String path = "";
+
+      // Check if we are using a file logger.
+      if (!(testObject.getLog() instanceof FileLogger)) {
+        // Since this is not a file logger we will need to use a generic file name
+        path = savePageSource(appiumDriver, testObject, LoggingConfig.getLogDirectory(),
+            StringProcessor.safeFormatter(
+               "%s - %s%s", "PageSource",
+                DateTimeFormatter.ofPattern(
+                    DEFAULT_DATE_TIME_FORMAT,
+                    Locale.getDefault()).format(LocalDateTime.now(Clock.systemUTC())),
+                appendName));
+      } else {
+        // Calculate the file name
+        String directory = ((FileLogger) testObject.getLog()).getDirectory();
+        String fileNameWithoutExtension = StringProcessor.safeFormatter(
+            "%s - %s%s", testObject.getFullyQualifiedTestName() + "_PS",
```

Comment:
SonarCloud mentions that `%s - %s%s` is repeated and should be a constant instead.  Valid?

Reply:
Technically yes, but it's a weird thing to put in a constant in my opinion. What do you think?

Reply:
Honestly, I'm indifferent, but I figured that it was worth the ask.

Reply:
Let's make sonarcloud happy.  put it as a logmessage prefix field and it'll probably get moved to utilities proper later.